xarray/plot/plot.py

```diff
@@ -221,18 +222,38 @@ def _infer_line_data(darray, x, y, hue):
 
         if y is None:
             xname, huename = _infer_xy_labels(darray=darray, x=x, y=hue)
-            yname = darray.name
-            xplt = darray.coords[xname]
-            yplt = darray.transpose(xname, huename)
+            xplt = darray[xname]
+            if xplt.ndim > 1:
+                if huename in darray.dims:
+                    otherdim = darray.dims[
+                        np.mod(darray.dims.index(huename) + 1, 2)]
+                    yplt = darray.transpose(otherdim, huename)
+                    xplt = xplt.transpose(otherdim, huename)
+                else:
+                    raise ValueError('If x or y are 2D, hue must be a dimension' +
+                                     'i.e. one of' + repr(darray.dims))
+
+            else:
+                yplt = darray.transpose(xname, huename)
 
         else:
             yname, huename = _infer_xy_labels(darray=darray, x=y, y=hue)
-            xname = darray.name
-            xplt = darray.transpose(yname, huename)
-            yplt = darray.coords[yname]
+            yplt = darray[yname]
+            if yplt.ndim > 1:
+                if huename in darray.dims:
+                    otherdim = darray.dims[
+                        np.mod(darray.dims.index(huename) + 1, 2)]
```

Comment:
I'm not sure to understand the call to ``np.mod`` here. This is going to work with 2D arrays only, right?

Reply:
yup, addressed.

Reply:
This should be the same as in the xcase (you changed only the block above).

Note that the same could be achieved with ``otherdim = darray.dims[int(darray.dims.index(huename) == 0)]`` (maybe less explicit).